src/functions/fetch.ts

```diff
@@ -0,0 +1,29 @@
+import { getUserByUsername, insertUser } from "../modules/db"
+import { getUser } from "../modules/github"
+import { User } from "../types"
+
+// Function to retrieve and store user from Github
+export const fetchUser = async (username: string): Promise<User> => {
+  // Check if user exists in DB
+  let user = await getUserByUsername(username)
+
+  if (!user) {
```

Comment:
Wouldn't this be simpler with

```
if (user) {
  return user;
}
... <the rest of the logic>
```

This way the `user` variable can even be a `const` instead of a mutable variable

Reply:
Yes, this solution is simpler indeed. The other one could help with DRY if we want to change the user before returning it, but it's obviously not needed in this case.